code/modules/client/preference/preferences.dm

```diff
@@ -2230,8 +2189,35 @@ GLOB.glassesblock
 		else if(firstspace == name_length)
 			real_name += "[pick(GLOB.last_names)]"
 
-	/* Disabilities, Oh My Freakin' Heck, Y'all are Killing Voxxy */
-	character.add_language(language)			//Add Secondary Language to list of languages our character can speak
+	character.add_language(language) //Add Secondary Language to list of languages our character can speak
+
+	/*
+	Disabilities,
+	Oh My Freakin' Heck,
+	Y'all are Killing Voxxy
+	*/
+
+	var/list/dblock_list = new(GLOB.glassesblock, GLOB.colourblindblock, GLOB.blindblock, GLOB.deafblock, GLOB.muteblock, GLOB.fatblock, GLOB.swedeblock, GLOB.chavblock, GLOB.lispblock, GLOB.clumsyblock)
+	for(var/D in 1 to dname_list.len)	//Let's use THOSE lists!
+		var/disflag = dflag_list[D]
+		if(disabilities & disflag)
+			var/dna_type = DNA_SE	//Curable Disabilities go onto the DNA_SE list
+			var/discure = dcure_list[D]
+			if(disflag == DISABILITY_FLAG_FAT)
+				character.overeatduration = 600
+			if(disabilities_cures & discure)
+				dna_type = DNA_RP	//Incurable Disabilities go onto the DNA_RP list
```

Comment:
Now I have to build out DNA_RP to be more like DNA_SE. Or, at least get it to a length that it can accept all the possible disabilities we can set in Preferences x3

@blinkdog I thought this would be easier than trying to set up a new system of a toggling a `cure` hook, in this code area, on the disability genes and then scripting around that wherever those genes were added/removed (mutations, viruses, cluwning, admerey, etc.).

Reply:
wait. shit. this means....I have to get the....genes to trigger....as well...right? crap. I don't know.

Either way, I've built out the toggle.